### PR TITLE
reduce cloudfront cost (move collection processing to server-side)

### PR DESCRIPTION
**Pull request by adriangohjw**
## Problem

Cloudfront cost should not be as high based on our current traffic

A sizeable portion comes from serving huge collection layout

## Solution

**Breaking Changes**

- [ ] Yes - this PR contains breaking changes
  - Details ...
- [x] No - this PR is backwards compatible

**Improvements**:

move as much client-side rendering to server-side to be processed during build time. This reduces the need to
- pass in the huge `sitemap`
- fields that are not displayed but only used as an intermediary variable to compute final output 
- similar approach to https://github.com/opengovsg/isomer/pull/848 in general

using `moh-corp` archived repo of just 200+ collection items, this reduces the HTML file from 1.28MB to 520kb (40% reduction). With larger site like actual moh-corp PROD, it should be much more significant.

And with current CF actual usage, quick napkin math says it can reduce by at least 20% (will be much more)

note - shifted stories from CollectionCard to Collection

### CODE REVIEW

**Comment by adriangohjw** [comment]
| | Before | After | Note |
| - | - | - | - |
| Uncompressed (S3) | 10.7MB | 3.9MB | -64% reduction |
| Compressed (CF) | 11.2MB | 488kB | -96% reduction<br>No idea why compressed is larger prior |
| Time taken to load | 378ms | 48ms | -87% reduction |

**Review by seaerchin — COMMENTED** [review]
i don't have an issue with this PR and i am ok to approve it, however i don't quite understand _why_ this helps! 

i thought we would precompute on the server side and this is done through removing `site` and getting the final output for `imageSrc` and `imageTitle`?

File: packages/components/src/interfaces/internal/CollectionCard.ts [hunk 1]
```diff
@@ -33,7 +33,20 @@ export interface LinkCardProps extends BaseCardProps {
...
+  "lastUpdated" | "category" | "title" | "description" | "image"
+> & {
+  referenceLinkHref: string | undefined
+  imageSrc: string | undefined
+  itemTitle: string
+}
+
+// NOTE: This is to ensure no additional props are being passed to this component
+export type ProcessedCollectionCardProps = CollectionCardProps &
+  Record<string, never>
```
[the remark this answers is not in the document]
Comment (adriangohjw) [reply packages/components/src/interfaces/internal/CollectionCard.ts:52]: it's derived from `CollectionCardProps` which uses `Pick` and from what i've read, `Pick` doesn't strictly enforce additional props from being passed in.

File: packages/components/src/templates/next/layouts/Collection/Collection.tsx [hunk 2]
```diff
@@ -102,7 +104,42 @@ const getCollectionItems = (
...
+        url,
+        site.siteMap,
+        site.assetsBaseUrl,
+      ),
+      imageSrc:
+        isExternalUrl(item.image?.src) || site.assetsBaseUrl === undefined
+          ? item.image?.src
+          : `${site.assetsBaseUrl}${item.image?.src}`,
+      itemTitle: `${item.title}${file ? ` [${file.type.toUpperCase()}, ${file.size.toUpperCase()}]` : ""}`,
+    } as ProcessedCollectionCardProps // Ensure no additional props are being returned
```
[the remark this answers is not in the document]
Comment (seaerchin) [reply packages/components/src/templates/next/layouts/Collection/Collection.tsx:141]: just tried, i tihnk just using `satisfies` (without `Exact`) should suffice!

File: packages/components/src/templates/next/layouts/Collection/Collection.tsx [hunk 3]
```diff
@@ -102,7 +104,42 @@ const getCollectionItems = (
...
+      referenceLinkHref: getReferenceLinkHref(
+        url,
+        site.siteMap,
+        site.assetsBaseUrl,
+      ),
+      imageSrc:
+        isExternalUrl(item.image?.src) || site.assetsBaseUrl === undefined
+          ? item.image?.src
+          : `${site.assetsBaseUrl}${item.image?.src}`,
+      itemTitle: `${item.title}${file ? ` [${file.type.toUpperCase()}, ${file.size.toUpperCase()}]` : ""}`,
```
[the remark this answers is not in the document]
Comment (adriangohjw) [reply packages/components/src/templates/next/layouts/Collection/Collection.tsx:140]: possibly but this is the previous logic which I didn't really wanna touch

[the remark this answers is not in the document]
Comment (seaerchin) [reply packages/components/src/templates/next/layouts/Collection/Collection.tsx:142]: ok lets keep it since it isn't a regression

**Pull request merged**